pages/dao/[symbol]/proposal/components/instructions/UXD/EditCredixLpDepository.tsx

```diff
@@ -0,0 +1,182 @@
+import Switch from '@components/Switch';
+import { useState } from 'react';
+import createEditCredixLpDepositoryInstruction from '@tools/sdk/uxdProtocol/createEditCredixLpDepositoryInstruction';
+
+const schema = yup.object().shape({
+  governedAccount: yup
+    .object()
+    .nullable()
+    .required('Governance account is required'),
+  collateralName: yup.string().required('Valid Collateral name is required'),
+  mintingFeeInBps: yup
+    .number()
+    .min(0, 'Minting fee in bps should be min 0')
+    .max(255, 'Minting fee in bps should be max 255'),
+  redeemingFeeInBps: yup
+    .number()
+    .min(0, 'Redeeming fee in bps should be min 0')
+    .max(255, 'Redeeming fee in bps should be max 255'),
+  uiRedeemableAmountUnderManagementCap: yup
+    .number()
+    .min(0, 'Redeemable amount under management cap should be min 0'),
+});
+
+const RegisterCredixLpDepository = ({
+  index,
+  governedAccount,
+}: {
+  index: number;
+  governedAccount?: GovernedMultiTypeAccount;
+}) => {
+  const [mintingFeesInBpsChange, setMintingFeesInBpsChange] = useState<boolean>(
+    false,
+  );
+
+  const [
+    redeemingFeesInBpsChange,
+    setRedeemingFeesInBpsChange,
+  ] = useState<boolean>(false);
+
+  const [
+    uiRedeemableAmountUnderManagementCapChange,
+    setUiRedeemableAmountUnderManagementCapChange,
+  ] = useState<boolean>(false);
+
+  const {
+    connection,
+    form,
+    formErrors,
+    handleSetForm,
+  } = useInstructionFormBuilder<UXDEditCredixLpDepositoryForm>({
+    index,
+    initialFormValues: {
+      governedAccount,
+    },
+    schema,
+    buildInstruction: async function ({ form, governedAccountPubkey }) {
+      return createEditCredixLpDepositoryInstruction({
+        connection,
+        uxdProgramId: form.governedAccount!.governance!.account.governedAccount,
+
+        authority: governedAccountPubkey,
+
+        // TODO
+        // Temporary authority override for tests with mainnet test program
+        // authority: new PublicKey(
+        //  '8cJ5KH2ExX2rrY6DbzAqrBMDkQxYZfyedB1C4L4osc5N',
+        // ),
+        // ================
+
+        depositoryMintName: form.collateralName!,
+        mintingFeeInBps: mintingFeesInBpsChange
+          ? form.mintingFeeInBps!
+          : undefined,
+
+        redeemingFeeInBps: redeemingFeesInBpsChange
+          ? form.redeemingFeeInBps!
+          : undefined,
+
+        redeemableAmountUnderManagementCap: uiRedeemableAmountUnderManagementCapChange
+          ? form.uiRedeemableAmountUnderManagementCap!
+          : undefined,
+      });
+    },
+  });
+
+  return (
+    <>
+      <Select
+        label="Collateral Name"
+        value={form.collateralName}
+        placeholder="Please select..."
+        onChange={(value) =>
+          handleSetForm({ value, propertyName: 'collateralName' })
+        }
+        error={formErrors['collateralName']}
+      >
+        <SelectOptionList list={getDepositoryMintSymbols(connection.cluster)} />
```

Comment:
Credix is only USDC for now and most likely also in the future, you could simplify all form by removing Collateral selection

Reply:
Yeah agreed, I was thinking about this too, but devnet has a special collateral that is different from mercurial, i was unsure what to do as I didnt want to hardcode too much things